### PR TITLE
[docs] update on_after_backward example

### LINKED ISSUE #5525: on_after_backward docs for logging histograms
#### What is your question?

I was following the example `on_after_backward` code for logging parameter variables for tensorboard histograms.
https://pytorch-lightning.readthedocs.io/en/latest/_modules/pytorch_lightning/core/hooks.html#ModelHooks.on_after_backward

#### Code

    # example to inspect gradient information in tensorboard
    if self.trainer.global_step % 25 == 0:  # don't make the tf file huge
        params = self.state_dict()
        for k, v in params.items():
            self.logger.experiment.add_histogram(
                tag=k, values=v.grad, global_step=self.trainer.global_step
                )

and I was getting errors with `v.grad` in my `state_dict` `v.grad` is returning NoneTypes rather than valid tensors for all tensors?

Has the API for pytorch changed perhaps? or is this method mainly just suitable for logging the values of the variables instead of the gradients?

#### What's your environment?

 - OS: Ubuntu 20.04
 - Packaging pip
 - Version 1.1.4
pytorch 1.7.0

### PR DESCRIPTION
**Pull request by rohitgr7**
## What does this PR do?

Fixed #5525 (just a minor doc update)

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [ ] Did you write any new necessary tests? (not for typos and docs)
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified
 - [x] **Check that target branch and milestone match!**

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by carmocca — APPROVED** [review]

**Review by s-rog — APPROVED** [review]

**Review by SkafteNicki — APPROVED** [review]

**Review by kaushikb11 — APPROVED** [review]

**Comment by Borda** [comment]
@rohitgr7  there won't be any more releases in 1.1.x so I have changed your destination branch to feat 1.2
pls, there is no need to close this PR, just rebase on the target branch, or maybe easier just cherry-pick the commit (@kaushikb11 will help you if needed)

**Comment by edenlightning** [comment]
@rohitgr7 looks like this needs a fix

**Comment by Borda** [comment]
ok, that was an issue with switching branches, for while protection was off...
